### PR TITLE
[react-select] Fix Control component styles config function

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/JedWatson/react-select/blob/master/packages/react-select/src/Select.js#L805
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-select ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-select@latest/))
was missing the following properties:
1. defaultTheme
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a6c49aa0706c2b3a812e81e692ec24f2ada300c3
</p>

File: types/react-select/src/styles.d.ts [hunk 1]
```diff
@@ -1,63 +1,35 @@
...
-  multiValueLabelCSS,
-  multiValueRemoveCSS,
-} from './components/MultiValue';
 import { CSSProperties } from 'react';
+import { ControlProps } from './components/Control';
 
 export interface Props { [key: string]: any; }
 
-export type styleFn = (base: CSSProperties, state: any) => CSSProperties;
+export type StylesConfigFunction<Props = any> = (base: CSSProperties, props: Props) => CSSProperties;
```
Comment (andrewbranch) [inline types/react-select/src/styles.d.ts:6]: Maybe consider leaving `styleFn` in and `/*@deprecated*/` it to ease the transition for anyone who might be importing it now, but I leave it up to you.

**Review by andrewbranch — APPROVED** [review]

**Comment by Methuselah96** [comment]
Ready to merge

**Comment by Methuselah96** [comment]
If it hadn't taken 2.5 weeks to get this merged, then I would make the suggest change, but it's not worth waiting any longer.

**Pull request merged**